components/Feature/Services/index.js

```diff
@@ -78,16 +85,34 @@ const Services = ({
     };
   };
 
+  const showAllClicked = e => {
+    setShowMoreResults(!showMoreResults);
+    sendDataToAnalytics({
+      action: getUserGroup(referrerData['user-groups']),
+      category: SHOW_MORE_RESULTS,
+      label: resultsTitle,
+      value: filteredResources.resources.length
+    });
+  };
+
   const handleSearch = e => {
     e.preventDefault();
     setFeedbackSubmitted(false);
+    setShowMoreResults(false);
 
     const searchTerm = e.target['search-input'].value;
     setResultsTitle(searchTerm);
 
-    const searchResults = getSearchResults(searchTerm, categorisedResources);
-    const newFilteredResources = flattenSearchResults(searchResults);
-    newFilteredResources.resources.sort((a, b) => b.weight - a.weight);
+    const filteredByCategory = filterByCategories(selectedCategories, categorisedResources);
+
+    const searchResults = getSearchResults(searchTerm, filteredByCategory);
+
+    if (selectedCategories.length > 0) weightByCategories(selectedCategories, categorisedResources);
```

Comment:
so from what I gather, weightByCategories returns a new JSON, but we're not assigning it to anything?

Reply:
This clearly isn't right. I've reverted back to how it should be.